app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java

```diff
@@ -56,84 +56,78 @@
     private int toDay;
     private int toMonth;
     private int toYear;
-    private DatabaseHandler dB;
-    private MainActivity activity;
-
-    public ExportPresenter(MainActivity exportActivity, DatabaseHandler dbHandler) {
-        this.activity = exportActivity;
-        dB = dbHandler;
-    }
 
-    public void onExportClicked(final Boolean all) {
-        final List<GlucoseReading> readings;
+    private Activity mActivity;
+    private ExportView mExportView;
+    private DatabaseHandler dB;
 
-        if (all) {
-            readings = dB.getGlucoseReadings();
+    public ExportPresenter(Activity activity) {
+        mActivity = activity;
+        if (activity instanceof ExportView) {
+            mExportView = (ExportView) activity;
         } else {
-            Calendar fromDate = Calendar.getInstance();
-            fromDate.set(Calendar.YEAR, fromYear);
-            fromDate.set(Calendar.MONTH, fromMonth);
-            fromDate.set(Calendar.DAY_OF_MONTH, fromDay);
-
-            Calendar toDate = Calendar.getInstance();
-            toDate.set(Calendar.YEAR, toYear);
-            toDate.set(Calendar.MONTH, toMonth);
-            toDate.set(Calendar.DAY_OF_MONTH, toDay);
-            readings = dB.getGlucoseReadings(fromDate.getTime(), toDate.getTime());
+            throw new RuntimeException("ExportPresenter Activity must implement ExportView interface");
         }
+        dB = GlucosioApplication.getInstance().getDBHandler();
+    }
+
+    public void onExportClicked(final boolean isExportAll) {
+
+        if (hasStoragePermissions(mActivity)) {
+//            final ReadingToCSV csv = new ReadingToCSV(GlucosioApplication.getInstance());
+            final String preferredUnit = dB.getUser(1).getPreferred_unit();
+            final boolean[] isEmpty = {false};
```

Comment:
Would it be better to create simple structure {String //filename, boolean //isEmpty} instead of having several states to update?